src/test/functions/env.spec.ts

```diff
@@ -0,0 +1,164 @@
+import { expect } from "chai";
+import * as sinon from "sinon";
+
+import { FirebaseError } from "../../error";
+import * as fenv from "../../functions/env";
+import * as envstore from "../../functions/envstore";
+
+describe("function env", () => {
+  describe("parseKvArgs", () => {
+    it("should successfully parse kv args", () => {
+      const args = ["FOO=bar", "BAR=foo=bar"];
+      expect(fenv.parseKvArgs(args)).to.deep.equal({ FOO: "bar", BAR: "foo=bar" });
+    });
+
+    it("should throw error given invalid keys", () => {
+      const args = ["FOO=bar", "X_GOOGLE_BAR=foo=bar"];
+      expect(() => {
+        fenv.parseKvArgs(args);
+      }).to.throw(FirebaseError);
+    });
+  });
+
+  describe("validateKey", () => {
+    it("should accept valid keys", () => {
+      const keys = ["FOO", "ABC_EFG", "A1_B2"];
+      keys.forEach((key) => {
+        expect(() => {
+          fenv.validateKey(key);
+        }).not.to.throw;
+      });
+    });
+
+    it("should throw error given invalid keys", () => {
+      const keys = ["", "A", "1F", "B=C"];
```

Comment:
Nit: Isn't it impossible for you to have the key `B=C` because the parser will first split at the `=` and consider only `B`?